### PR TITLE
More docs

### PR DESCRIPTION
**Pull request by jdlrobson**

### CODE REVIEW

File: MediaWiki.d.ts [hunk 1]
```diff
@@ -87,14 +100,16 @@ interface MwEventLog {
...
 interface mwHookInstance {
-	add( fn: Function ): this;
+	add( fn:  mwHookFunction ): this;
 	/**
 	 * Fire an event for logging.
-	 *
-	 * @param [data] The data describing the event
 	 */
-	fire( data?: any ): void;
+	fire( ...args: any[] ): void;
```
Comment (nicholasray) [inline MediaWiki.d.ts:112]: It looks like this method returns `this` as well?

File: MediaWiki.d.ts [hunk 2]
```diff
@@ -147,10 +162,14 @@ interface MwUser {
 	sessionId(): string;
 }
 
+interface MwLogger {
+	warn( ...args: any[] ): void;
```
Comment (nicholasray) [inline MediaWiki.d.ts:166]: I think this is specified to only accept any number of string types [1]

[1] https://github.com/wikimedia/mediawiki/blob/5c573b3e3abfac20dd417087352816e57c7c4294/resources/src/startup/mediawiki.js#L169

File: MediaWiki.d.ts [hunk 3]
```diff
@@ -70,6 +82,7 @@ type MwApiConstructor = new( options?: Object ) => MwApi;
 
 interface MwCookie {
 	get( cookieName: string ): () => string
+	set( cookieName: string, value: string|null ): () => void;
```
Comment (nicholasray) [inline MediaWiki.d.ts:85]: It looks like this is specified as taking an optional third param [1]

[1] https://github.com/wikimedia/mediawiki/blob/5c573b3e3abfac20dd417087352816e57c7c4294/resources/src/mediawiki.cookie/index.js#L60-L64

File: MediaWiki.d.ts (hunk 2 above)
Comment (jdlrobson) [reply MediaWiki.d.ts:166]: `mw.log.warn({})` is valid too.. so I think this is right?

File: MediaWiki.d.ts (hunk 3 above)
Comment (jdlrobson) [reply MediaWiki.d.ts:85]: Done.

File: MediaWiki.d.ts [hunk 4]
```diff
@@ -168,7 +187,12 @@ interface MediaWiki {
 	echo?: MwEcho,
 	eventLog?: MwEventLog,
 	experiments: MwExperiments;
+	log: MwLogger;
 	util: {
+		/**
+		 * @param {string} selector
+		 */
+		getTargetFromFragment( selector: string ): HTMLElement|null;
```
[the remark this answers is not in the document]
Comment (jdlrobson) [reply MediaWiki.d.ts:195]: Done.

File: MediaWiki.d.ts (hunk 1 above)
Comment (jdlrobson) [reply MediaWiki.d.ts:112]: Done.

File: MediaWiki.d.ts (hunk 2 above)
Comment (nicholasray) [reply MediaWiki.d.ts:176]: Yeah, I'm not sure why the JSDoc documents it as limited to only strings, but I'm fine with what you've written

**Review by nicholasray — APPROVED** [review]

**Pull request merged**